sound/soc/intel/boards/sof_realtek_common.c

```diff
@@ -0,0 +1,138 @@
+// SPDX-License-Identifier: GPL-2.0-only
+//
+// Copyright(c) 2020 Intel Corporation. All rights reserved.
+
+#include <linux/kernel.h>
+#include <sound/pcm.h>
+#include <sound/pcm_params.h>
+#include <sound/soc.h>
+#include <sound/soc-dai.h>
+#include <sound/soc-dapm.h>
+#include <uapi/sound/asound.h>
+#include "../../codecs/rt1011.h"
+#include "sof_realtek_common.h"
+
+/*
+ * Current only 2-amp configuration is supported for rt1011
+ */
+static const struct snd_soc_dapm_route rt1011_dapm_routes[] = {
+	/* speaker */
+	{ "Left Spk", NULL, "Left SPO" },
+	{ "Right Spk", NULL, "Right SPO" },
+};
+
+/*
+ * Make sure device's Unique ID follows this configuration:
+ *
+ * Two speakers:
+ *         0: left, 1: right
+ * Four speakers:
+ *         0: Woofer left, 1: Woofer right
+ *         2: Tweeter left, 3: Tweeter right
+ */
+static struct snd_soc_codec_conf rt1011_codec_confs[] = {
+	{
+		.dlc = COMP_CODEC_CONF(RT1011_DEV0_NAME),
+		.name_prefix = "Left",
+	},
+	{
+		.dlc = COMP_CODEC_CONF(RT1011_DEV1_NAME),
+		.name_prefix = "Right",
+	},
+};
+
+static struct snd_soc_dai_link_component rt1011_dai_link_components[] = {
+	{
+		.name = RT1011_DEV0_NAME,
+		.dai_name = RT1011_CODEC_DAI,
+	},
+	{
+		.name = RT1011_DEV1_NAME,
+		.dai_name = RT1011_CODEC_DAI,
+	},
+};
+
+struct {
```

Comment:
can this be static const? please make sure you compile with
make W=1 sound/soc/intel/boards/
make C=2 sound/soc/intel/boards/

Reply:
W=1 is ok but C=2 fails to compile...

brentlu@brentlu-brix:/workspace/sof/linux$ make C=2 sound/soc/intel/boards/
  CHECK   scripts/mod/empty.c
/bin/sh: 1: sparse: not found
scripts/Makefile.build:283: recipe for target 'scripts/mod/empty.o' failed
make[1]: *** [scripts/mod/empty.o] Error 127
Makefile:1199: recipe for target 'prepare0' failed
make: *** [prepare0] Error 2

Reply:
sparse is a tool you install, seems from your error messages that you don't have it.

Reply:
Seems it crashes with or without my patch...

brentlu@brentlu-brix:/workspace/sof/linux$ make C=2 sound/soc/intel/boards/
  CHECK   scripts/mod/empty.c
<...>
./include/linux/restart_block.h:30:29: error: undefined identifier '__user'
./include/linux/restart_block.h:30:29: error: too many errors
Segmentation fault (core dumped)
scripts/Makefile.build:283: recipe for target 'sound/soc/intel/boards/sof_rt5682.o' failed
make[4]: *** [sound/soc/intel/boards/sof_rt5682.o] Error 139
scripts/Makefile.build:500: recipe for target 'sound/soc/intel/boards' failed
make[3]: *** [sound/soc/intel/boards] Error 2
scripts/Makefile.build:500: recipe for target 'sound/soc/intel' failed
make[2]: *** [sound/soc/intel] Error 2
scripts/Makefile.build:500: recipe for target 'sound/soc' failed
make[1]: *** [sound/soc] Error 2
Makefile:1799: recipe for target 'sound' failed
make: *** [sound] Error 2